File: src/main/java/com/github/joelws/release/tracker/service/artifact/execution/ReadArtifactServiceExecution.java
package com.github.joelws.release.tracker.service.artifact.execution;

import com.github.joelws.release.tracker.dao.artifact.ArtifactDao;
import com.github.joelws.release.tracker.entity.artifact.Artifact;
import com.github.joelws.release.tracker.service.ServiceExecution;
import org.apache.log4j.Logger;

public class ReadArtifactServiceExecution implements ServiceExecution<Artifact, Artifact>
{
    private static final Logger LOGGER = Logger.getLogger(ReadArtifactServiceExecution.class);

    private final ArtifactDao artifactDao;

    public ReadArtifactServiceExecution(final ArtifactDao artifactDao)
    {
        this.artifactDao = artifactDao;
    }

    @Override public Artifact execute(Artifact param)
    {
        LOGGER.info("Finding Artifact with the following: " +
                param.getId().getGroupId() + " " +
                param.getId().getArtifactId() + " " +
                param.getId().getVersion());

        return artifactDao.read(param.getId());
    }
}

File: src/main/java/com/github/joelws/release/tracker/service/artifact/execution/ReadArtifactServiceExecution.kt
package com.github.joelws.release.tracker.service.artifact.execution

import com.github.joelws.release.tracker.dao.artifact.ArtifactDao
import com.github.joelws.release.tracker.entity.artifact.Artifact
import com.github.joelws.release.tracker.service.ServiceExecution
import org.apache.log4j.Logger

open class ReadArtifactServiceExecution(val artifactDao: ArtifactDao) : ServiceExecution<Artifact, Artifact?> {


    companion object {
        private var LOGGER = Logger.getLogger(ReadArtifactServiceExecution::class.java)
    }

    override fun execute(param: Artifact): Artifact? {
        LOGGER.info("Finding artifact with the following: ${param.id?.groupId}, ${param.id?.artifactId}, ${param.id?.version}")

        return artifactDao.read(param.id)
    }
}
